smurfs/src/reducer/smurfReducer.js

```diff
@@ -0,0 +1,50 @@
+import { START_FETCHING, FETCH_SUCCESS, FETCH_FAILURE, POST_SUCCESS } from '../actions'
+
+export const initialState = {
+  smurfs: [
+    // name: "Brainey", 
+    // age: 200, 
+    // height: "5cm", 
+    // id: 0
+  ],
+  isFetching: false,
+  error: ''
+}
+
+export const reducer = (state = initialState, action) => {
+  switch(action.type){
+    case START_FETCHING:
+      return{
+        ...state,
+        isFetching: true,
+        error: ''
+      };
```

Comment:
If we had multiple action types like `START_FETCHING` and `START_POSTING`, our reducer would look something like this:
```
case START_FETCHING:
    return {
        ...state,
        isFetching: true,
        error: ''
    }
case START_POSTING:
    return {
        ...state,
        isFetching: true,
        error: ''
    }
```

This code, however, is very redundant. To make it less repetitive, we could instead type:
```
case START_POSTING:
case START_FETCHING:
    return {
        ...state,
        isFetching: true,
        error: ''
    }
```
This way, both action types lead to the same block of code '' '